closed/src/java.base/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java

```diff
@@ -260,6 +260,21 @@ int encrypt(byte[] plain, int plainOffset, int plainLen,
             throw new ProviderException("Internal error in input buffering");
         }
 
+        /**
+         * OpenSSL doesn't support overlapping buffers.
+         * If plaintext and ciphertext byte arrays are overlapping, make a copy of plain
+         */
+        byte[] copyOfInput = null;
+        if (plain == cipher) {
+            if ((plainOffset >= cipherOffset) ||
+                ((plainOffset < cipherOffset) && (cipherOffset < (plainOffset + plainLen)))) {
+                copyOfInput = new byte[plainLen];
+                System.arraycopy(plain,  plainOffset, copyOfInput, 0, plainLen);
+                plain = copyOfInput;
+                plainOffset = 0;
+            }
+        }
```

Comment:
The declaration of `copyOfInput` should move into the `if` statement.

Is it really worth checking the offsets (`encryptFinal()` does not)? If so:
* the overlap test includes a redundant check (`a || (!a && b))` is just `a || b`)
* the overlap test is overly strong when `plainOffset >= cipherOffset` (there's no overlap if `plainOffset + plainLen >= cipherOffset`)

Reply:
updated.